lean/elf.lean

```diff
@@ -531,6 +590,27 @@ def pp_phdrs {c:elf_class} (phdrs:list (phdr c)) : io unit := do
     io.put_str_ln $ "Index:            " ++ idx.repr,
     io.put_str_ln phdr.snd.pp)
 
+def region := char_buffer
+
+def elfmem : Type := data.imap ℕ region
+
+def read_one_elfmem {c : elf_class} (m : elfmem) (ph : phdr c) : file_input elfmem :=
+  match ph.phdr_type with
+  | phdr_type.PT_LOAD := do
+    file_input.seek ph.offset.val,
+    fbs <- file_input.read (min ph.filesz.val ph.memsz.val),
+    monad_lift (io.put_str_ln ("read_one_elfmem: read " ++ fbs.size.repr ++ " bytes")),
+    let val := buffer.append_list fbs (list.repeat (char.of_nat 0) (ph.memsz.val - ph.filesz.val)) 
+    in pure $ data.imap.insert ph.vaddr.val ph.memsz.val val m
+  | _       := pure m
+  end
```

Comment:
Could you add a comment on what this and region/elfmem definitions are intended to do?